tests/deploy_tests/test_updates.py

```diff
@@ -0,0 +1,730 @@
+        """
+        return app_store_descriptor.TankAppStoreDescriptor(
+            None, None, {"name": name, "type": "app_store", "version": version}, bundle_type
+        )
+
+    def __init__(self, pc_path, bundle_install_path, location_dict, bundle_type):
+        """
+        Constructor.
+        """
+        AppDescriptor.__init__(self, pc_path, bundle_install_path, location_dict)
+        self._type = bundle_type
+
+    def get_system_name(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        return self.get_location()["name"]
+
+    def get_version(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        return self.get_location()["version"]
+
+    def _get_metadata(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        return {
+            "frameworks": MockStore.instance.get_bundle(
+                self._type, self.get_system_name(), self.get_version()
+            ).required_frameworks
+        }
+
+    def run_post_install(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        pass
+
+    def exists_local(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        return True
+
+    @classmethod
+    def find_latest_item(cls, pc_path, bundle_install_path, bundle_type, name, constraint_pattern=None):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        if constraint_pattern:
+            return cls._find_latest_for_pattern(bundle_type, name, constraint_pattern)
+        else:
+            return cls._find_latest(bundle_type, name)
+
+    def find_latest_version(self, version_pattern=None):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        if version_pattern:
+            return self._find_latest_for_pattern(
+                self._type,
+                self.get_system_name(),
+                version_pattern
+            )
+        else:
+            return self._find_latest(
+                self._type,
+                self.get_system_name()
+            )
+
+    @classmethod
+    def _find_latest(cls, bundle_type, name):
+        """
+        Finds the latest version of a bundle.
+
+        :param bundle_type: Type of the bundle. Any of AppDescriptor.{APP,ENGINE,FRAMEWORK}
+        :param name: Name of the bundle.
+
+        :returns: A MockStore._Entry descriptor for the latest version of the bundle.
+        """
+
+        versions = MockStore.instance.get_bundle_versions(bundle_type, name)
+        latest = "v0.0.0"
+        for version in versions:
+            if LooseVersion(version) > LooseVersion(latest):
+                latest = version
+        return cls.create(name, latest, bundle_type)
+
+    @classmethod
+    def _find_latest_for_pattern(cls, bundle_type, name, version_pattern):
```

Comment:
now i am worried this may have to change with the 0.18 refactors. Probably not a lot, but still. Probably makes sense if you do the merge into the 0.18 branch @jfboismenu to make sure it's done in the best possible way.